src/Common/AzurePSCmdlet.cs

```diff
@@ -331,6 +342,18 @@ protected virtual void SetupHttpClientPipeline()
 
         protected virtual void TearDownHttpClientPipeline()
         {
+            try
+            {
+                string hostEnv = AzurePSCmdlet.getEnvUserAgent();
+                if (!String.IsNullOrWhiteSpace(hostEnv))
+                {
+                    AzureSession.Instance.ClientFactory.RemoveUserAgent(hostEnv);
+                }
+            }
+            catch (Exception)
```

Comment:
I add a catch here just in case the env var is not OK for setting UserAgent, we'll ignore it.